pwndbg/commands/spray.py

```diff
@@ -52,7 +54,27 @@ def spray(addr, length, value) -> None:
     else:
         value_bytes = cyclic(length, n=pwndbg.gdblib.arch.ptrsize)
 
-    try:
-        pwndbg.gdblib.memory.write(addr, value_bytes)
-    except gdb.MemoryError as e:
-        print(M.error(e))
+    if only_funcptrs:
+        try:
+            mem = pwndbg.gdblib.memory.read(addr, length)
+        except gdb.MemoryError as e:
+            print(M.error(e))
+            return
+
+        addresses_written = 0
+        ptrsize = pwndbg.gdblib.arch.ptrsize
```

Comment:
Can we move this to the begining of the whole function? There is a `pwndbg.gdblib.arch.ptrsize` used in line 53 that could re-use this :P